dhall/CHANGELOG.md

```diff
@@ -1,3 +1,86 @@
+* BREAKING CHANGE TO THE API: [Treat multi-`let`s as syntactic sugar](https://github.com/dhall-lang/dhall-haskell/pull/1242)
+    * This is a breaking change because the `Let` constructor now only stores
+      one `Binding` instead of a `NonEmpty` list of `BInding`s
+* TECHNICALLY BREAKING CHANGE TO THE LANGUAGE: [Dependent types](https://github.com/dhall-lang/dhall-haskell/pull/1164)
+    * You can now write functions from terms to types
+    * There is also now language support for tests of the form
+      `assert : x === y`
+    * This is a technically breaking change because `assert` is now a reserved
+      keyword
+    * See the [changelog for standard version 10.0.0](https://github.com/dhall-lang/dhall-lang/releases/tag/v10.0.0) for more details
+* TECHNICALLY BREAKING CHANGE TO THE LANGUAGE: [Add `Natural/subtract` built-in](https://github.com/dhall-lang/dhall-haskell/pull/1133)
+    * The language now supports machine subtraction, which can be used to
+      support several other high-performance operations (like `Natural`
+      comparisons)
+    * This is a technically breaking change if you used `Natural/subtract` as an
+      identifier in your code
+    * See the [changelog for standard version 10.0.0](https://github.com/dhall-lang/dhall-lang/releases/tag/v10.0.0) for more details
+* TECHNICALLY BREAKING CHANGE TO THE LANGUAGE: [More simplifications for field selection](https://github.com/dhall-lang/dhall-haskell/pull/1174)
+    * Now the interpreter will more intelligently simplify certain field
+      projections
+    * For example: `λ(x : { a : Bool, b : Bool }) → (x ⫽ { c = 0 }).{ a, c }.c`
+      will now simplify to `λ(x : { a : Bool, b : Bool }) → 0 `
+    * This is a technically breaking change because you will need to update
+      integrity checks that protect code simplified in this way
+    * See the [changelog for standard version 10.0.0](https://github.com/dhall-lang/dhall-lang/releases/tag/v10.0.0) for more details
+* TECHNICALLY BREAKING CHANGE TO THE LANGUAGE: [Simplify `⫽` when its arguments are equivalent](https://github.com/dhall-lang/dhall-haskell/pull/1196)
+    * This is a technically breaking change for the same reason: this will
+      perturb semantic integrity checks for affected code
+    * See the [changelog for standard version 10.0.0](https://github.com/dhall-lang/dhall-lang/releases/tag/v10.0.0) for more details
+* NEW FEATURE: [Restore support for records containing both types and terms](https://github.com/dhall-lang/dhall-haskell/pull/1173)
+    * In other words `{ foo = 1, bar = Bool }` is now valid again
+    * This means that you now can export a single package containing both types
+      and terms
+    * See the [changelog for standard version 10.0.0](https://github.com/dhall-lang/dhall-lang/releases/tag/v10.0.0) for more details
+* [`dhall format` now preserves `let` comments](https://github.com/dhall-lang/dhall-haskell/pull/1273)
```

Comment:
This is causing a breaking change, because `Binding` has been altered. (The new `Binding` is kind of obscure, as well; some docs would go a long way.)

I'll put together a PR for more docs and some other improvements to the new `Binding`.